ygen/ir.go

```diff
@@ -338,23 +358,65 @@ const (
 
 // YANGNodeDetails stores the YANG-specific details of a node
 // within the schema.
+// TODO(wenbli): Split this out so that parts can be re-used by
+// ParsedDirectory.
 type YANGNodeDetails struct {
 	// Name is the name of the node from the YANG schema.
 	Name string
 	// Defaults represents the 'default' value(s) directly
 	// specified in the YANG schema.
 	Defaults []string
-	// Module stores the name of the module that instantiates
-	// the node.
-	Module string
-	// Path specifies the absolute YANG schema node path.
+	// BelongingModule is the module in whose namespace the node belongs.
+	BelongingModule string
+	// RootModule is the module in which the YANG tree the node is attached
+	// to was instantiated (rather than the module that has the same
+	// namespace as the node).
+	//
+	// In this example, leaf has a RootModule of "openconfig-simple", while a
+	// BelongingModule of "openconfig-augment". The DefiningModule is
+	// "openconfig-augment".
+	//
+	//   module openconfig-augment {
+	//     import openconfig-simple { prefix "t"; }
+	//     augment "/t:parent/child/state" {
+	//       leaf zero { type string; }
+	//     }
+	//   }
+	RootModule string
```

Comment:
Yes it is indeed used for determining the package name as well as just documenting to the user which module instantiates the root of the tree containing it https://github.com/openconfig/ygot/pull/657#discussion_r859294556

Reply:
See comment in #657.

Reply:
This is a good idea, although the problem here is it's hinged on us never augmenting a top-level node, because if we do then the module in the path (belonging/instantiating module) will be different than the root module. Since `RootModule` here is the same for **all nodes under a tree**, then I'm thinking it makes sense for this to be a separate field as is now. I'm in favour of removing the `Path` field instead after we figure out the right way to key `ParsedDirectory`.